lib/src/main/java/com/thenewboston/api/confirmationvalidatorapi/datasource/ConfirmationDataSource.kt

```diff
@@ -102,4 +104,9 @@ class ConfirmationDataSource @Inject constructor(
         call = { postDataSource.doSendPrimaryValidatorUpdated(request) },
         errorMessage = "Could not send primary validator update notice"
     )
+
+    suspend fun sendUpgradeRequest(request: UpgradeRequest): Outcome<ValidatorDetails> = makeApiCall(
```

Comment:
We need to include the validator details in a new data class with the json structure defined in the documentation. This request will fail in the real environment. 

https://thenewboston.com/confirmation-validator-api/upgrade-request

The "primary_validator" node can be used for the ValidatorDetails data class along with the rest of the properties described in the json below:

``json
{
  "primary_validator": {
    "account_number": "2e86f48216567302527b69eae6c6a188097ed3a9741f43cc3723e570cf47644c",
    "ip_address": "54.183.17.224",
    "node_identifier": "2262026a562b0274163158e92e8fbc4d28e519bc5ba8c1cf403703292be84a51",
    "port": 1337,
    "protocol": "http",
    "version": "v1.0",
    "default_transaction_fee": 1,
    "root_account_file": "http://54.183.17.224/media/root_account_file.json",
    "root_account_file_hash": "cc9390cc579dc8a99a1f34c1bea5d54a0f45b27ecee7e38662f0cd853f76744d",
    "seed_block_identifier": "",
    "daily_confirmation_rate": null,
    "trust": 100.0
  },
  "account_number": "ad1f8845c6a1abb6011a2a434a079a087c460657aad54329a84b406dce8bf314",
  "ip_address": "54.219.178.46",
  "node_identifier": "b82a2fac55dd39f667a53157c68100ba856d07272f21e62b54155a5200415ff6",
  "port": 1337,
  "protocol": "http",
  "version": "v1.0",
  "default_transaction_fee": 1,
  "root_account_file": "http://70.79.52.191:1337/media/root_account_file.json",
  "root_account_file_hash": "cc9390cc579dc8a99a1f34c1bea5d54a0f45b27ecee7e38662f0cd853f76744d",
  "seed_block_identifier": "",
  "daily_confirmation_rate": null,
  "node_type": "PRIMARY_VALIDATOR"
}
``

Reply:
The `ValidatorDetails` class has the same json structure:
![image](https://user-images.githubusercontent.com/50385769/110558599-c2a53980-8110-11eb-9fde-ef87766b02f7.png)


And the `Validator` Class:
![image](https://user-images.githubusercontent.com/50385769/110558688-ecf6f700-8110-11eb-837f-8ad09ca92813.png)

Reply:
You're right my bad. I didn't read the `primaryValidator ` property in `ValidatorDetails`